src/main/java/com/bravo/user/utility/ValidatorUtil.java

```diff
@@ -41,12 +41,27 @@ else if(value instanceof DateFilter<?>){
     else if(value instanceof String){
       isValid = isStringValid((String)value);
     }
+    else if(value instanceof Integer) {
+      isValid = isIntegerValid((Integer)value);
+    }
     else {
       isValid = value != null;
     }
     return isValid;
   }
 
+  public static <T> boolean isInvalidLength(T value, T length){
+    return !isValidLength(value, length);
+  }
+
+  public static <T> boolean isValidLength(T value, T length){
+    boolean isValid = false;
+    if(value instanceof Integer){
+       isValid = isIntegerLengthValid((Integer) value, (Integer) length);
+      }
+    return isValid;
+  }
```

Comment:
I don't like the implementation of this method. It's not very generic. It would be better if it took a min and max range and then compared the value for numbers and the length for strings.


     public static <T> boolean isValidLength(T value, double min, double max){
          Double numeric;
          if(value instanceof Number){
               numeric = ((Number)value).doubleValue();          
          }
          else if(value instanceof String){
               numeric =(double)((String)value).length;
          }
          else {
               throw new IllegalArgumentException("helpful message");
          }
          return numeric > min && numeric < max;
     }

Reply:
I was thinking in a purely linear way here in regards to testing only Integer values. With that in mind, I am not sure if using a range where I would have to set both a min and max  arg to the same value to test the ceiling of an Integer value would be more effective than my isValidLength function which when all told should handle checks on only Integers in less code than your re-factored example(unless your point was to emphasize additional data types being covered).